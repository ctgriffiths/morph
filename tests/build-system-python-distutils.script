#!/bin/sh
#
# Convert the hello-chunk project to python with distutils
#
# Copyright (C) 2011, 2012  Codethink Limited
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; version 2 of the License.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.


set -eu

chunkrepo="$DATADIR/chunk-repo"
cd "$chunkrepo"

git checkout --quiet farrokh

git rm --quiet hello.c
cat <<EOF >hello
#!/usr/bin/python
print "hello, world"
EOF
git add hello

cat <<EOF >setup.py
#!/usr/bin/python
from distutils.core import setup
setup(name='hello',
      scripts=['hello'])
EOF
git add setup.py

cat <<EOF >hello.morph
{
    "name": "hello",
    "kind": "chunk",
    "build-system": "python-distutils"
}
EOF
git add hello.morph

git commit --quiet -m 'convert hello into a python project'

"$SRCDIR/scripts/test-morph" build --prefix= chunk-repo farrokh hello.morph

for chunk in "$DATADIR/cache/artifacts/"*.chunk.*
do
    echo "$chunk:" | sed 's/[^.]*//'
    tar -tf "$chunk" | LC_ALL=C sort | sed '/^\.\/./s:^\./::'
    echo
done |
sed -e 's:^local/::' \
    -e 's:lib/python2.[6-9]/:lib/python2.x/:' \
    -e 's:/hello-0\.0\.0[^/]*\.egg-info$:/hello.egg-info/:' \
    -e 's:[^/]*-packages:packages:' \
    -e '/^$/d'
