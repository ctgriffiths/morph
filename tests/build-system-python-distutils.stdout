.chunk.hello:
./
baserock/
baserock/hello.meta
bin/
bin/hello
lib/
lib/python2.x/
lib/python2.x/dist-packages/
lib/python2.x/dist-packages/hello.egg-info/
